src/modules/commander/Commander.cpp

```diff
@@ -165,6 +165,7 @@ static uint64_t rc_signal_lost_timestamp;		// Time at which the RC reception was
 
 static uint8_t arm_requirements = ARM_REQ_NONE;
 
+static bool _last_condition_local_altitude_valid = false;
```

Comment:
This is fine as a quick change to slip into the next release, but let's re-evaluate later. We're trying to get rid of the static variables and really we probably don't even need to carry additional state when the validity condition is updated earlier in the same loop.

Reply:
I'm aware of this and I totally agree, we should also revisit the reconsideration of the RC mode switch. There are some corner cases especially when you gain GPS lock in air that we need to get rid of. I'm open to look into it after the release.

Reply:
Some quick hacks we can use to discuss this later. https://github.com/PX4/Firmware/compare/master...dagar:fix-rc-mode-altitude-on-boot?expand=1